test/launcher.spec.js

```diff
@@ -31,30 +31,65 @@ module.exports.addTests = function({testRunner, expect, defaultBrowserOptions, p
-        let revisionInfo = browserFetcher.revisionInfo('123456');
+        const expectedRevision = '123456';
+        let revisionInfo = browserFetcher.revisionInfo(expectedRevision);
         server.setRoute(revisionInfo.url.substring(server.PREFIX.length), (req, res) => {
           server.serveFile(req, res, '/chromium-linux.zip');
         });
 
         expect(revisionInfo.local).toBe(false);
         expect(browserFetcher.platform()).toBe('linux');
+        expect(browserFetcher.product()).toBe('chrome');
+        expect(!!browserFetcher.host()).toBe(true);
         expect(await browserFetcher.canDownload('100000')).toBe(false);
-        expect(await browserFetcher.canDownload('123456')).toBe(true);
+        expect(await browserFetcher.canDownload(expectedRevision)).toBe(true);
 
-        revisionInfo = await browserFetcher.download('123456');
+        revisionInfo = await browserFetcher.download(expectedRevision);
         expect(revisionInfo.local).toBe(true);
         expect(await readFileAsync(revisionInfo.executablePath, 'utf8')).toBe('LINUX BINARY\n');
         const expectedPermissions = os.platform() === 'win32' ? 0666 : 0755;
         expect((await statAsync(revisionInfo.executablePath)).mode & 0777).toBe(expectedPermissions);
-        expect(await browserFetcher.localRevisions()).toEqual(['123456']);
-        await browserFetcher.remove('123456');
+        expect(await browserFetcher.localRevisions()).toEqual([expectedRevision]);
+        await browserFetcher.remove(expectedRevision);
+        expect(await browserFetcher.localRevisions()).toEqual([]);
+        await rmAsync(downloadsFolder);
+      });
+    });
+    describe('BrowserFetcher', function() {
+      it('should download and extract firefox linux binary', async({server}) => {
+        const downloadsFolder = await mkdtempAsync(TMP_FOLDER);
+        const browserFetcher = puppeteer.createBrowserFetcher({
+          platform: 'linux',
+          path: downloadsFolder,
+          host: server.PREFIX,
+          product: 'firefox',
+        });
+        const expectedVersion = '75';
+        let revisionInfo = browserFetcher.revisionInfo(expectedVersion);
+        server.setRoute(revisionInfo.url.substring(server.PREFIX.length), (req, res) => {
+          server.serveFile(req, res, `/firefox-${expectedVersion}.0a1.en-US.linux-x86_64.tar.bz2`);
+        });
+
+        expect(revisionInfo.local).toBe(false);
+        expect(browserFetcher.platform()).toBe('linux');
+        expect(browserFetcher.product()).toBe('firefox');
+        expect(await browserFetcher.canDownload('100000')).toBe(false);
+        expect(await browserFetcher.canDownload(expectedVersion)).toBe(true);
+
+        revisionInfo = await browserFetcher.download(expectedVersion);
+        expect(revisionInfo.local).toBe(true);
+        expect(await readFileAsync(revisionInfo.executablePath, 'utf8')).toBe('FIREFOX LINUX BINARY\n');
+        const expectedPermissions = os.platform() === 'win32' ? 0666 : 0755;
```

Comment:
I realize this is just matching the existing convention throughout this file, but since this is new code: let's not use legacy octal literals, since they are invalid in strict mode / modules (which we might want to move to at some point). We can use explicit octal integer literals instead:

```suggestion
        const expectedPermissions = os.platform() === 'win32' ? 0o666 : 0o755;
```